lib/ReactViews/Map/Panels/SharePanel/SharePanel.tsx

```diff
@@ -0,0 +1,667 @@
+        this.props.viewState.shareModalIsVisible = true;
+      }
+    }
+  }
+
+  getShareUrlInput(theme: string) {
+    return (
+      <Input
+        className={Styles.shareUrlfield}
+        light={theme === "light"}
+        dark={theme === "dark"}
+        large
+        type="text"
+        value={this.state.shareUrl}
+        placeholder={this.state.placeholder}
+        readOnly
+        onClick={e => e.currentTarget.select()}
+        id="share-url"
+      />
+    );
+  }
+
+  getShareUrlInputStory(theme: string) {
+    return (
+      <Input
+        className={Styles.shareUrlfield}
+        light={theme === "light"}
+        dark={theme === "dark"}
+        large
+        type="text"
+        value={this.state.shareUrl}
+        placeholder={this.state.placeholder}
+        readOnly
+        onClick={e => e.currentTarget.select()}
+        id="share-url"
+        css={`
+          border-radius: 32px 0 0 32px;
+        `}
+      />
+    );
+  }
+
+  onAddWebDataClicked() {
+    // TODO: Make sure that this.setState is accessible, may need to this.updateForShortening = this.updateForShortening.bind(this) or change to arrow function
+    this.setState({
+      isOpen: false
+    });
+    this.props.viewState.openUserData();
+  }
+
+  hasUserAddedData() {
+    return this.props.terria.catalog.userAddedDataGroup.members.length > 0;
+  }
+
+  onIncludeStoryInShareClicked(event: React.ChangeEvent<HTMLInputElement>) {
+    this.setState({ includeStoryInShare: event.target.checked }, () => {
+      this.updateForShortening();
+    });
+  }
+
+  renderWarning() {
+    const unshareableItems = this.props.terria.catalog.userAddedDataGroup.memberModels.filter(
+      model => !isShareable(this.props.terria)(model.uniqueId || "")
+    );
+
+    return (
+      <>
+        {unshareableItems.length > 0 && (
+          <div className={Styles.warning}>
+            <Trans i18nKey="share.localDataNote">
+              <p className={Styles.paragraph}>
+                <strong>Note:</strong>
+              </p>
+              <p className={Styles.paragraph}>
+                The following data sources will NOT be shared because they
+                include data from this local system. To share these data
+                sources, publish their data on a web server and{" "}
+                <a
+                  className={Styles.warningLink}
+                  onClick={this.onAddWebDataClicked}
+                >
+                  add them using a url
+                </a>
+                .
+              </p>
+            </Trans>
+            <ul className={Styles.paragraph}>
+              {unshareableItems
+                .filter(CatalogMemberMixin.isMixedInto)
+                .map((item: CatalogMemberMixin.Instance, i) => {
+                  return (
+                    <li key={i}>
+                      <strong>{item.name}</strong>
+                    </li>
+                  );
+                })}
```

Comment:
It might be better to use the `getName` function (in `CatalogMemberMixin.ts` file) here - just in case we get some `BaseModels` which aren't using `CatalogMemberMixin`

```suggestion
              {unshareableItems.map((item, i) => {
                return (
                  <li key={i}>
                    <strong>{getName(item)}</strong>
                  </li>
                );
              })}
```

Reply:
Ah awesome, thats better. Done.